.github/workflows/composite/setup/action.yaml

```diff
@@ -0,0 +1,32 @@
+name: "Setup dependencies & build"
+description: "Checkout, setup dependencies and build go binaries"
+inputs:
+  build:
+    description: "Build"
+    default: true
+  setup-node-e2e:
+    description: "Setup node for e2e tests"
+    default: true
+  setup-foundry:
+    description: "Setup foundry"
+    default: true
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-go@v5
+      with:
+        go-version: "1.22"
+        cache: false
+    - uses: actions/setup-node@v4
+      if: "${{ inputs.setup-node-e2e == 'true' }}"
+      with:
+        node-version-file: "e2e_test/js-tests/package.json"
+    - name: Install Foundry
+      if: "${{ inputs.setup-foundry == 'true' }}"
+      uses: foundry-rs/foundry-toolchain@v1.4.0
+      with:
+        version: v1.1.0
+    - name: Build
```

Comment:
Not sure if we need this, it's only enabled for the test job, but the test target included the build target.